### PR TITLE
xds: move balancergroup and weightedtarget our of xds directory

### PR DESCRIPTION
**Pull request by easwars**
This change enables `balancergroup` package to be used outside of `xds`.

- Majority of the changes are in tests. Existing `balancergroup` tests were very closely tied to `weightedtarget`. With this change, the tests which test functionality in the latter have been moved to that package.
- The remainder of the `balancergroup` tests still use the state aggregator from `weightedtarget`. But these tests mainly tests functionality in the `balancergroup` package and it was less painful to move `weightedtarget` out of xds to enable this, rather than write a mock state aggregator (since that would have been non-trivial).
- Also, ended up moving a lot of testutils out of xds.

RELEASE NOTES: none

### CODE REVIEW

File: balancer/weightedtarget/weightedtarget_config_test.go [hunk 1]
```diff
@@ -22,35 +22,40 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+
```
Comment (menghanl) [inline balancer/weightedtarget/weightedtarget_config_test.go:25]: Delete this empty line

File: balancer/weightedtarget/weightedtarget_test.go [hunk 2]
```diff
@@ -0,0 +1,1221 @@
...
+package weightedtarget
+
+import (
+	"encoding/json"
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/google/go-cmp/cmp"
+
```
Comment (menghanl) [inline balancer/weightedtarget/weightedtarget_test.go:28]: Delete this line

File: xds/internal/balancer/cdsbalancer/cdsbalancer_security_test.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ import (
 	"testing"
 
 	"github.com/google/go-cmp/cmp"
+
```
Comment (menghanl) [inline xds/internal/balancer/cdsbalancer/cdsbalancer_security_test.go:27]: Delete

**Review by menghanl — APPROVED** [review]
LGTM. All comments are nits

File: balancer/weightedtarget/weightedtarget_config_test.go (hunk 1 above)
Comment (easwars) [reply balancer/weightedtarget/weightedtarget_config_test.go]: Done

File: balancer/weightedtarget/weightedtarget_test.go (hunk 2 above)
Comment (easwars) [reply balancer/weightedtarget/weightedtarget_test.go]: Done

File: xds/internal/balancer/cdsbalancer/cdsbalancer_security_test.go (hunk 3 above)
Comment (easwars) [reply xds/internal/balancer/cdsbalancer/cdsbalancer_security_test.go]: Done

**Pull request merged**